### PR TITLE
cleanup issues found during testing

### PR DESCRIPTION
**Pull request by malaffoon**

### CODE REVIEW

File: validation/README.md [hunk 1]
```diff
@@ -35,7 +38,7 @@ reporting_password=
 
 reporting_olap_host=localhost
 reporting_olap_port=5439
-reporting_olap_schema=reporting_olap
+reporting_olap_db=dev
```
Comment (malaffoon) [inline validation/README.md]: we've changed our sharing strategy so it is the db and user that is important in redshift

File: validation/README.md [hunk 2]
```diff
@@ -53,90 +56,92 @@ reporting_olap_password=
 ./validate-migration.sh secrets/local.sh olap
 ```
 
-NOTE: When running tests you may see multiple "Warning: Using a password on the command line interface can be insecure.", please ignore it.
```
Comment (malaffoon) [inline validation/README.md]: found a way to suppress that warning

File: validation/README.md [hunk 3]
```diff
@@ -53,90 +56,92 @@ reporting_olap_password=
 ./validate-migration.sh secrets/local.sh olap
 ```
 
-NOTE: When running tests you may see multiple "Warning: Using a password on the command line interface can be insecure.", please ignore it. 
+To investigate a failed test, open the printed directory link in the terminal output.
```
Comment (malaffoon) [inline validation/README.md]: just moved this up higher in the document

File: validation/README.md [hunk 4]
```diff
@@ -53,90 +56,92 @@ reporting_olap_password=
...
 |__) |  \ |  |     |\/| | / _` |__)  /\   |  | /  \ |\ |    \  /  /\  |    | |  \  /\   |  /  \ |__) 
 |  \ |__/ |/\|     |  | | \__> |  \ /~~\  |  | \__/ | \|     \/  /~~\ |___ | |__/ /~~\  |  \__/ |  \ 
 
-
 validating...
-
-  warehouse connection:      localhost:3306:warehouse:root
-  reporting connection:      localhost:3306:reporting:root
-  reporting olap connection: localhost:5439:reporting_olap:root
+  output folder: /tmp/validation/results-2018-01-19-143702
```
Comment (malaffoon) [inline validation/README.md]: tweaked output to provide a bit more info, inc. timestamps

File: validation/README.md [hunk 5]
```diff
@@ -53,90 +56,92 @@ reporting_olap_password=
...
-completed in 00:10:37
-```
-To investigate a failed test, open the printed directory link in the terminal output.
-If the comparison between warehouse and reporting_olap failed for instance, you would see the following files:
-```
-validation/results-2018-01-08-091053/total-ica-scores
-├── reporting.csv
-├── warehouse.csv
-└── warehouse_reporting.diff
+completed in 00:00:07
```
Comment (malaffoon) [inline validation/README.md]: added more info to summary

File: validation/validate-migration.sh [hunk 6]
```diff
@@ -8,14 +8,6 @@ function print_banner() {
     echo $'|__) |  \ |  |     |\/| | / _` |__)  /\   |  | /  \ |\ |    \  /  /\  |    | |  \  /\   |  /  \ |__) '
     echo $'|  \ |__/ |/\|     |  | | \__> |  \ /~~\  |  | \__/ | \|     \/  /~~\ |___ | |__/ /~~\  |  \__/ |  \ '
     echo $''
-    echo $''
-}
-
-function not_null() {
```
Comment (malaffoon) [inline validation/validate-migration.sh]: this wasn't doing what you might think; an exit called in a function doesn't exit the main script

File: validation/validate-migration.sh [hunk 7]
```diff
@@ -30,34 +22,18 @@ function format_seconds_to_HH_MM_SS() {
     echo `date -u -r $1 +%T`
 }
 
-function get_absolute_path() {
```
Comment (malaffoon) [inline validation/validate-migration.sh]: unused

File: validation/validate-migration.sh [hunk 8]
```diff
@@ -30,34 +22,18 @@ function format_seconds_to_HH_MM_SS() {
...
 function get_property_by_index() {
     echo $1 | cut -f${2} -d"|"
 }
 
 function print_connection() {
     declare -a connection=("${!1}")
-    echo "${connection[0]}:${connection[1]}:${connection[2]}:${connection[3]}"
-}
-
-function create_mysql_password_file() {
```
Comment (malaffoon) [inline validation/validate-migration.sh]: unused

File: validation/validate-migration.sh [hunk 9]
```diff
@@ -30,34 +22,18 @@ function format_seconds_to_HH_MM_SS() {
...
-    local file_path=`mktemp`
-    echo -e "[client]\npassword=${password}" > ${file_path} && chmod 600 ${file_path}
-    echo ${file_path}
+    echo "${connection[3]} @ ${connection[0]}:${connection[1]}/${connection[2]}"
 }
 
 function mysql_to_csv() {
     declare -a connection=("${!1}")
-    mysql -h ${connection[0]} -P ${connection[1]} -u ${connection[3]} ${connection[2]} -p${connection[4]} -s < ${sql_file} | tr '\t' ','
+    mysql -h ${connection[0]} -P ${connection[1]} -u ${connection[3]} ${connection[2]} -p${connection[4]} -s < ${sql_file} 2>&1 | grep -v "Warning: Using a password" | tr '\t' ','
```
Comment (malaffoon) [inline validation/validate-migration.sh]: trick to munch warning

File: validation/validate-migration.sh [hunk 10]
```diff
@@ -69,7 +45,7 @@ function run_tests() {
     declare -a tests=("${!1}")
     for test in "${tests[@]}"
     do
-        run_test_and_compare_restults ${test} $2 $3 $4 $5
+        run_test_and_compare_results ${test} $2 $3 $4 $5
```
Comment (malaffoon) [inline validation/validate-migration.sh]: typo in function name

File: validation/validate-migration.sh [hunk 11]
```diff
@@ -100,40 +76,26 @@ function compare_test_results() {
 
     diff ${diff_options} ${a} ${b} > ${comparison_file}
 
-    local total_differences=$(( `get_line_count ${comparison_file}` ))
+    local total_differences=$(( `cat ${comparison_file} | wc -l` ))
+
+    mkdir -p ${test_result_dir}
```
Comment (malaffoon) [inline validation/validate-migration.sh]: always emit data results, even if no differences

File: validation/validate-migration.sh [hunk 12]
```diff
@@ -100,40 +76,26 @@ function compare_test_results() {
...
         mv ${comparison_file} ${diff_file}
-        echo "  ${a_namespace}/${b_namespace} (${total_differences} differences) ${test_result_dir}"
+        diff_files+=(${diff_file})
+        echo "  ${a_namespace}/${b_namespace} (${total_differences} differences) ${diff_file}"
     fi
 }
 
 # setting dependent methods
 
-function validate_input() {
```
Comment (malaffoon) [inline validation/validate-migration.sh]: similar issue: the script is not being exited when there are missing configuration values

File: validation/validate-migration.sh (hunk 9 above)
Comment (esotrope) [reply validation/validate-migration.sh]: haxx : j

**Pull request merged**